kite-tools-parent/kite-tools/src/main/java/org/kitesdk/cli/commands/DeleteCommand.java

```diff
@@ -46,11 +51,24 @@ public int run() throws IOException {
         View view = Datasets.load(uriOrName);
         Preconditions.checkArgument(viewMatches(view.getUri(), uriOrName),
             "Resolved view does not match requested view: " + view.getUri());
-        view.deleteAll();
+
+        if(console.isDebugEnabled()){
+          for(PartitionView<?> pv: (Iterable<PartitionView<?>>) view.getCoveringPartitions()){
+            console.debug("Deleting partition {}", pv.getLocation());
```

Comment:
This assumes that the underlying view is a FS view and that it is aligned with partition boundaries. If either of those isn't true then this will either throw a ClassCastException or lie about what would have been deleted. We could add a `canDelete()` method to the view if necessary or just special case partitions?